### PR TITLE
Make `Odb` `Send` and `Sync`

### PR DESCRIPTION
**Pull request by joshtriplett**
As of libgit2 1.2.0, `git_odb` uses locking internally, and should be thread-safe. Mark it `Send` and `Sync` to allow access from multiple threads.

### CODE REVIEW

**Comment by joshtriplett** [comment]
One day I will get the marker-trait impl syntax right on the first try. But it is not this day.

**Comment by joshtriplett** [comment]
Also see https://github.com/libgit2/libgit2/pull/6109

**Review by alexcrichton — APPROVED** [review]
Seems good to me, but let's wait on merging until the upstream docs are merged?

**Comment by joshtriplett** [comment]
@alexcrichton It has now been merged.

**Pull request merged**
